middle_end/flambda2/to_cmm/to_cmm.ml

```diff
@@ -1301,13 +1301,23 @@ and make_arm ~tag_discriminant env res (d, action) =
    scrutinee is adequately tagged/untagged) *)
 and make_switch ~tag_discriminant env res e arms =
   let wrap, env = Env.flush_delayed_lets env in
-  match Targetint_31_63.Map.cardinal arms with
+  match Targetint_31_63.Lmap.bindings arms with
   (* Binary case: if-then-else *)
-  | 2 -> (
-    let aux = make_arm ~tag_discriminant env in
-    let first_arm, res = aux res @@ Targetint_31_63.Map.min_binding arms in
-    let second_arm, res = aux res @@ Targetint_31_63.Map.max_binding arms in
-    match first_arm, second_arm with
+  | [first_arm; second_arm] -> (
+    let first_arm_has_lower_discriminant =
```

Comment:
Why not simply sort the list using the discriminant as keys ? I suspect the code would be shorter and more readable, and at this point we know the list has length `2`, it's not costly.